### PR TITLE
Create unified views of VPN apps' Glean data (DENG-210)

### PR DESCRIPTION
**Pull request by sean-rose**
All the VPN apps used to use Glean.js and all that data was available in the `mozilla_vpn` dataset, but now the VPN Android app uses the Glean Kotlin SDK and its Glean data isn't included in the `mozilla_vpn` dataset.  The VPN team would like all the VPN apps' Glean data to be available together ([DENG-210](https://mozilla-hub.atlassian.net/browse/DENG-210)).

This PR:
- Changes VPN event funnel ETLs in the `mozilla_vpn_derived` dataset to include the VPN Android app data.
- Changes `mozilla_vpn.events_unnested` view to include VPN Android app data.
- Stops generating `mozilla_vpn_android.events_unnested` view (it doesn't return any data anyway).
- Overrides the default `main`, `deletion_request`, and `events` views in the `mozilla_vpn` dataset to union the VPN apps' associated Glean ping data together.

### Deployment plan:
1. Merge PR and wait for CI to complete.
2. Re-run `bqetl_event_rollup.mozilla_vpn_derived__event_types_history__v1` Airflow tasks from 2022-06-22 to date (2022-06-22 is when the earliest separate VPN Android app Glean data arrived).
3. Re-run latest `bqetl_event_rollup.mozilla_vpn_derived__event_types__v1` Airflow task.
4. Re-run `bqetl_event_rollup.mozilla_vpn_derived__events_daily__v1` Airflow tasks from 2022-06-22 to date.
5. Drop `moz-fx-data-shared-prod.mozilla_vpn_android.events_unnested` view.

---
Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title)
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

### CODE REVIEW

**Comment by sean-rose** [comment]
> Creates `mozilla_vpn.all_main` and `mozilla_vpn.all_deletion_request` views that union the VPN apps' associated Glean ping data together.

I'd prefer to have those just be `mozilla_vpn.main` and `mozilla_vpn.deletion_request`, but such views are currently being generated which only select from the same views in the `mozillavpn` (no underscore) dataset.

Is there some way to prevent the current `mozilla_vpn.main` and `mozilla_vpn.deletion_request` views from being generated?

File: sql/moz-fx-data-shared-prod/mozilla_vpn_derived/events_daily_v1/query.sql [hunk 1]
```diff
@@ -29,6 +29,36 @@ WITH sample AS (
...
+    (
+      SELECT
+        ARRAY_AGG(STRUCT(key, value.branch AS value))
+      FROM
+        UNNEST(ping_info.experiments)
+    ) AS experiments
+  FROM
+    org_mozilla_firefox_vpn.main e
+  CROSS JOIN
+    UNNEST(e.events) AS event
```
Comment (scholtzan) [inline sql/moz-fx-data-shared-prod/mozilla_vpn_derived/events_daily_v1/query.sql:61]: Do these two tables `events_daily_v1` and `event_type_history_v1` need to get backfilled?

**Review by scholtzan — APPROVED** [review]

**Comment by relud** [comment]
> Is there some way to prevent the current mozilla_vpn.main and mozilla_vpn.deletion_request views from being generated?

I believe that when a view is manually defined then generation will skip that view (and thus use the manually defined one)

**Review by relud — APPROVED** [review]

File: sql/moz-fx-data-shared-prod/mozilla_vpn_derived/events_daily_v1/query.sql (hunk 1 above)
Comment (sean-rose) [reply sql/moz-fx-data-shared-prod/mozilla_vpn_derived/events_daily_v1/query.sql:61]: Yep!  From the deployment plan in the description:
> 2. Re-run `bqetl_event_rollup.mozilla_vpn_derived__event_types_history__v1` Airflow tasks from 2022-06-22 to date (2022-06-22 is when the earliest separate VPN Android app Glean data arrived).

> 4. Re-run `bqetl_event_rollup.mozilla_vpn_derived__events_daily__v1` Airflow tasks from 2022-06-22 to date.

**Comment by sean-rose** [comment]
> I believe that when a view is manually defined then generation will skip that view (and thus use the manually defined one)

[Right you are](https://github.com/mozilla/bigquery-etl/blob/5b20a0fe1cd4d7414563ec215ce3003d4fb45098/sql_generators/stable_views/__init__.py#L7-L9), thanks!

I'll rename the `all_main` and `all_deletion_request` views to override the default `main` and `deletion_request` views (which will make @birdsarah's life a bit easier).

File: sql/moz-fx-data-shared-prod/mozilla_vpn/events_unnested/view.sql [hunk 2]
```diff
@@ -24,6 +24,30 @@ SELECT
   event.name AS event_name,
   event.extra AS event_extra,
 FROM
-  `moz-fx-data-shared-prod.mozilla_vpn.main` AS e
+  `moz-fx-data-shared-prod.mozillavpn.main` AS e
```
[the remark this answers is not in the document]
Comment (sean-rose) [reply sql/moz-fx-data-shared-prod/mozilla_vpn/events_unnested/view.sql:27]: Glean has "app names" and "app IDs", where the app name can encompass one or more app IDs (typically for separate channels like release/beta/nightly).  The raw Glean data is synced into app ID-specific datasets, and then app name-specific datasets have aggregations and/or passthrough views pulling from the app ID datasets.

In this case `mozilla_vpn` (with underscore) is an app name dataset for the VPN clients using Glean.js (and is where we've also put a bunch of custom ETL views), and `mozillavpn` (without underscore) is an underlying app ID dataset for the same.

`mozilla_vpn.main` used to just be doing `SELECT * FROM mozillavpn.main` so this change was to cut out a middleman and keep this view's SQL more consistent with the auto-generated SQL in other ETL that was unioning data from the `mozillavpn` and `org_mozilla_firefox_vpn` app ID datasets.  However, now that we're overriding `mozilla_vpn.main` to do such unioning I think I'll actually simplify this `events_unnested` view to select from `mozilla_vpn.main` so it doesn't have to do any unioning itself.

**Comment by sean-rose** [comment]
> > I believe that when a view is manually defined then generation will skip that view (and thus use the manually defined one)
> 
> [Right you are](https://github.com/mozilla/bigquery-etl/blob/5b20a0fe1cd4d7414563ec215ce3003d4fb45098/sql_generators/stable_views/__init__.py#L7-L9), thanks!
> 
> I'll rename the `all_main` and `all_deletion_request` views to override the default `main` and `deletion_request` views (which will make @birdsarah's life a bit easier).

It turns out the `mozilla_vpn` ping views were generated by `glean_usage` (not `stable_views`), which doesn't avoid overwriting files.  I'm changing that in #3149, and then the unified `main` and `deletion_request` views from this PR should take effect.

**Pull request merged**